nixos/modules/services/x11/display-managers/lightdm.nix

```diff
@@ -72,17 +40,17 @@ let
       ${cfg.extraSeatDefaults}
     '';
 
-  gtkGreeterConf = writeText "lightdm-gtk-greeter.conf"
-    ''
-    [greeter]
-    theme-name = Adwaita
-    icon-theme-name = Adwaita
-    background = ${cfg.background}
-    '';
-
 in
 {
+  # Note: the order in which lightdm greeter modules are imported
+  # here determines the default: later modules (if enable) are
+  # preferred.
```

Comment:
First of, if another greeter gets enabled, it should disable the default greeter, similar to how if lightdm is enabled, it disables the default displayManager, slim.

However, if for some reason multiple greeters are enabled, the order of this import list will determine what greeter will be loaded, as every greeter should set the greeter option in this file. The last greeter to be loaded should overwrite the previously configuration in this option, and will thus be the one that is loaded.

As I'm new to nix/nixos I'm not a 100% sure that it works this way, but from what I gathered from other modules, it should. If this is not the way nix works, I'll have to find another way to define which greeter gets loaded by lightdm.